drivers/usb/udc/udc_nrf.c

```diff
@@ -140,6 +161,7 @@ static void udc_event_xfer_in(const struct device *dev,
 		buf = udc_buf_get(dev, ep, true);
 		if (buf == NULL) {
 			LOG_ERR("ep 0x%02x queue is empty", ep);
+			__ASSERT(false, "ep 0x%02x queue is empty", ep);
```

Comment:
Asserting on false is not that useful. It would be better to make it `__ASSERT(buf != NULL, "ep 0x%02x queue is empty", ep);`. Also I think it would be better to place assert before `if (buf == NULL)`